Add versioning to committee hooks

edisonywh opened the pull request:
Right now Committee will skip installation if a file contains the
identifier, `# committee` (per #25).  However this means that we won't be able to
change the implementation of the runner file in the future.

With versions, we can check the version to see if we should
regenerate the runner file.

With this commit, if the version field is not the same as the current
version, then we will regenerate/update the hooks. If the version field
is the same as the hooks however, we will skip.

edisonywh committed:
Add versioning to committee hooks

Right now Committee will skip installation if a file contains the
identifier, `# committee`. However this means that we won't be able to
change the implementation of the runner file.

With this version, we can check the version to see if we should
regenerate the runner file.

With this commit, if the version field is not the same as the current
version, then we will regenerate/update the hooks. If the version field
is the same as the hooks however, we will skip.

edisonywh commented:
@thiamsantos What do you think about this PR? This is what we talked about in #17 in regards to versioning.

I'm thinking whether or not we should also rename the backup to something different, so instead of a  generic `pre_commit.old`, since now we have a version we can maybe even backup to `pre_commit.version.old`, but need to think about restoring as well.

Anyway let me know what you think!

thiamsantos commented on lib/committee.ex:
```diff
@@ -1,9 +1,16 @@
+  def __hooks__, do: @hooks
+
+  @doc """
+  Return the current version of Committee
```
```suggestion
  Return the current version of Committee hook file format
```
WDYT? It will be different than the actual version of the package, right?

edisonywh replied:
Ah I see, yeah you're right it will be different. I'll update the copy

thiamsantos commented on lib/hooks.ex:
```diff
@@ -85,6 +107,7 @@ defmodule Committee.Hooks do
     #!/bin/sh
 
     #{@identifier}
+    # version:#{Committee.__version__()}
```
WDYT of using the identifier as marker for the version? To avoid we getting anything that has `version` in it.

`#{@identifier}:#{Committee.__version__()}`

edisonywh replied:
Yeah okay I see where you're coming from, I think you're right. I initially was thinking we could add more stuffs like dates or whatsoever while maintaining the same structure:

```
# committee
# version:1.0
# date:02-04-2020
```

But then again I don't think there's a lot of use for that, so I'm going to follow your suggestion. 👍

thiamsantos commented on test/committee_test.exs:
```diff
@@ -49,14 +49,44 @@ defmodule CommitteeTest do
     end
 
-    test "existing committee hooks", context do
+    test "outdated existing committee hooks should update to new version", context do
```
Should we also update files that were generated by committee, that don't contain the `version` in it? Or as we didn't published the changes, we are safe assuming that everyone will have the proper identifier and version in place.

edisonywh replied:
Yup, that was exactly where I was coming from - right now no one would have a githook file with identifier but without versions, so it's safe to push this out :)

thiamsantos commented:
> I'm thinking whether or not we should also rename the backup to something different, so instead of a generic pre_commit.old, since now we have a version we can maybe even backup to pre_commit.version.old, but need to think about restoring as well.

@edisonywh  Personally I think renaming to just `.old` is enough, we don't need to think in all the possibilities on that. Usually if someone willingly will set up a git hook tool, usually he assumes to everything be handled by it, and also that any hook that he previously had will be lost. So I think we are doing more than enough adding a possibility to backup and restore, even a simple one. Not all of the tools existing out there allow something like

edisonywh commented:
@thiamsantos Good point, having more complex backups seems a little unnecessary. 👍

edisonywh committed:
Update version identifier

As per [this PR](https://github.com/edisonywh/committee/pull/26), we are
deciding to change the version identifier from the following:

```
\# committee
\# version:1.0
```

to the following:

```
\# committee:1.0
```

This is because otherwise we will never be able to use the word
`version:` in the stubbed githook. It's probably unlikely that we ever
will need, but this is good measure to stick the version identifier
right after the identifier (it can't go wrong)

edisonywh commented:
I have updated as per suggestion, let me know what you think and then I'm going to push this to prod :)
